### PR TITLE
IDEMPIERE-5334 Restore revenue/expense accounts in charges

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5334

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/acct/Doc.java [hunk 1]
```diff
@@ -1303,8 +1303,10 @@ public int getValidCombination_ID (int AcctType, MAcctSchema as)
 			int cmp = getAmount(AMTTYPE_Charge).compareTo(Env.ZERO);
 			if (cmp == 0)
 				return 0;
-			else
+			else if (cmp < 0)
 				sql = "SELECT CH_Expense_Acct FROM C_Charge_Acct WHERE C_Charge_ID=? AND C_AcctSchema_ID=?";
+			else
+				sql = "SELECT CH_Revenue_Acct FROM C_Charge_Acct WHERE C_Charge_ID=? AND C_AcctSchema_ID=?";
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/acct/Doc.java:1309]: is this by sign logic always the safe thing to do ? perhaps a potential problem with reversal ?

Comment (hengsin) [reply org.adempiere.base/src/org/compiere/acct/Doc.java:1309]: ideally, we need unit test to validate the normal and corner cases

**Comment by hengsin** [comment]
Adjustment for handling of reversal and invoice posting.
[1379_reversal_patch.txt](https://github.com/idempiere/idempiere/files/9034146/1379_reversal_patch.txt)

Ignore this if you decided to close this PR instead.

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hengsin - as said in the ticket, at the end I think this ticket is not necessary.

I'm closing the pull request, but please feel free to create a new PR with your approach if you consider it worthy for core.

Regards,

Carlos Ruiz
